src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const sep = i < this.length - 1 ? separator : '';
```

Comment:
You can avoid this constant if you will use two sequential `if` blocks.
Smth like this:
```
if (this[i] !== null && this[i] !== undefined) {
  // do smth
}

if (I < this.length) {
  // do smth
}
```

You can try it on your own.